tket2-hseries/src/extension/wasm.rs

```diff
@@ -0,0 +1,885 @@
+    extension::{
+        prelude::{option_type, usize_t, PRELUDE_ID},
+        simple_op::{
+            try_from_name, HasConcrete, HasDef, MakeExtensionOp, MakeOpDef, MakeRegisteredOp,
+            OpLoadError,
+        },
+        ExtensionBuildError, ExtensionId, ExtensionRegistry, ExtensionSet, SignatureError,
+        SignatureFunc, TypeDefBound, Version, PRELUDE,
+    },
+    ops::{
+        constant::{downcast_equal_consts, CustomConst, ValueName},
+        ExtensionOp, NamedOp, OpName, OpType,
+    },
+    type_row,
+    types::{
+        type_param::{TypeArgError, TypeParam},
+        CustomType, FuncValueType, PolyFuncTypeRV, Signature, SumType, Type, TypeArg, TypeBound,
+        TypeEnum, TypeRV, TypeRow, TypeRowRV,
+    },
+    Extension, Wire,
+};
+use itertools::Itertools as _;
+use lazy_static::lazy_static;
+use serde::{Deserialize, Serialize};
+use smol_str::SmolStr;
+use strum::{EnumIter, EnumString, IntoStaticStr};
+
+use crate::extension::futures;
+
+use super::futures::future_type;
+
+/// The "tket2.wasm" extension id.
+pub const EXTENSION_ID: ExtensionId = ExtensionId::new_unchecked("tket2.wasm");
+/// The "tket2.wasm" extension version.
+pub const EXTENSION_VERSION: Version = Version::new(0, 1, 0);
+
+lazy_static! {
+    /// The `tket2.wasm` extension.
+    pub static ref EXTENSION: Arc<Extension> = Extension::new_arc(EXTENSION_ID, EXTENSION_VERSION, |ext, ext_ref| {
+        ext.add_requirements(ExtensionSet::from_iter([futures::EXTENSION_ID, PRELUDE_ID]));
+        add_wasm_type_defs(ext, ext_ref).unwrap();
+        WasmOpDef::load_all_ops(ext, ext_ref, ).unwrap();
+    });
+
+    /// A [Weak] reference to the `tket2.wasm` op.
+    pub static ref EXTENSION_REF: Weak<Extension> = Arc::downgrade(&EXTENSION);
+
+    /// Extension registry including the "tket2.wasm" extension and
+    /// dependencies.
+    pub static ref REGISTRY: ExtensionRegistry = ExtensionRegistry::new([
+        EXTENSION.to_owned(),
+        futures::EXTENSION.to_owned(),
+        PRELUDE.to_owned()
+    ]);
+
+    /// The name of the `tket2.wasm.module` type.
+    pub static ref MODULE_TYPE_NAME: SmolStr = SmolStr::new_inline("module");
+    /// The name of the `tket2.wasm.context` type.
+    pub static ref CONTEXT_TYPE_NAME: SmolStr = SmolStr::new_inline("context");
+    /// The name of the `tket2.wasm.func` type.
+    pub static ref FUNC_TYPE_NAME: SmolStr = SmolStr::new_inline("func");
+
+    /// The [TypeParam] of `tket2.wasm.lookup` specifying the name of the function.
+    pub static ref NAME_PARAM: TypeParam = TypeParam::String;
+    /// The [TypeParam] of various types and ops specifying the input signature of a function.
+    pub static ref INPUTS_PARAM: TypeParam = TypeParam::List { param: Box::new(TypeBound::Any.into()) };
+    /// The [TypeParam] of various types and ops specifying the output signature of a function.
+    pub static ref OUTPUTS_PARAM: TypeParam = TypeParam::List { param: Box::new(TypeBound::Any.into()) };
+}
```

Comment:
It might be that cargo fmt-ing this without the macro will break up some of the lines that are looking quite long at the moment.